stl/inc/algorithm

```diff
@@ -3342,13 +3366,246 @@ _FwdIt2 remove_copy_if(_ExPo&&, _FwdIt1 _First, _FwdIt1 _Last, _FwdIt2 _Dest, _P
+
+    // ALIAS TEMPLATE remove_copy_result
+    template <class _In, class _Out>
+    using remove_copy_result = in_out_result<_In, _Out>;
+
+    // VARIABLE ranges::remove_copy
+    class _Remove_copy_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <input_iterator _It, sentinel_for<_It> _Se, weakly_incrementable _Out, class _Ty, class _Pj = identity>
+            requires indirectly_copyable<_It, _Out>
+                  && indirect_binary_predicate<ranges::equal_to, projected<_It, _Pj>, const _Ty*>
+        constexpr remove_copy_result<_It, _Out> operator()(
+            _It _First, _Se _Last, _Out _Result, const _Ty& _Val, _Pj _Proj = {}) const {
+            _Adl_verify_range(_First, _Last);
+            auto _UResult = _Remove_copy_unchecked(_Get_unwrapped(_STD move(_First)), _Get_unwrapped(_STD move(_Last)),
+                _Get_unwrapped_unverified(_STD move(_Result)), _Val, _Pass_fn(_Proj));
+
+            _Seek_wrapped(_First, _STD move(_UResult.in));
+            _Seek_wrapped(_Result, _STD move(_UResult.out));
+            return {_STD move(_First), _STD move(_Result)};
+        }
+
+        template <input_range _Rng, weakly_incrementable _Out, class _Ty, class _Pj = identity>
+            requires indirectly_copyable<iterator_t<_Rng>, _Out>
+                  && indirect_binary_predicate<ranges::equal_to, projected<iterator_t<_Rng>, _Pj>, const _Ty*>
+        constexpr remove_copy_result<borrowed_iterator_t<_Rng>, _Out> operator()(
+            _Rng&& _Range, _Out _Result, const _Ty& _Val, _Pj _Proj = {}) const {
+            auto _First   = _RANGES begin(_Range);
+            auto _UResult = _Remove_copy_unchecked(_Get_unwrapped(_STD move(_First)), _Uend(_Range),
+                _Get_unwrapped_unverified(_STD move(_Result)), _Val, _Pass_fn(_Proj));
+
+            _Seek_wrapped(_First, _STD move(_UResult.in));
+            _Seek_wrapped(_Result, _STD move(_UResult.out));
+            return {_STD move(_First), _STD move(_Result)};
+        }
+        // clang-format on
+    private:
+        template <class _It, class _Se, class _Out, class _Ty, class _Pj>
+        _NODISCARD static constexpr remove_copy_result<_It, _Out> _Remove_copy_unchecked(
+            _It _First, const _Se _Last, _Out _Result, const _Ty& _Val, _Pj _Proj) {
+            // Copy [_First, _Last) to _Result except projected values equal to _Val
+            _STL_INTERNAL_STATIC_ASSERT(input_iterator<_It>);
+            _STL_INTERNAL_STATIC_ASSERT(sentinel_for<_Se, _It>);
+            _STL_INTERNAL_STATIC_ASSERT(weakly_incrementable<_Out>);
+            _STL_INTERNAL_STATIC_ASSERT(indirectly_copyable<_It, _Out>);
+            _STL_INTERNAL_STATIC_ASSERT(indirect_binary_predicate<ranges::equal_to, projected<_It, _Pj>, const _Ty*>);
+
+            for (; _First != _Last; ++_First) {
+                if (_STD invoke(_Proj, *_First) != _Val) {

```

Comment:
I'm not sure what "same question" you're referring to. My crystal ball says you want to know if the use of `!=` here is covered by the constraints. If so, the answer is yes*: `ranges::equal_to`  requires its arguments to model `equality_comparable_with` which requires both `==` and `!=` to be valid and have the typical meaning.

*This is a lie, but it is the design intent. As I touched on yesterday when we were discussing intrinsics for `BUILTIN_PTR_CMP` the point of the "or converts to pointers when compared" part of the constrained function object design is to punch a hole through the `equality_comparable` semantic requirement that substitutable values and only substitutable values compare equal. Since we want these function objects to totally order pointers, we have to allow for them to say that non-substitutable pointers are equal (say, a pointer past-the-end of an object that happens to represent the same address as a non-past-the-end pointer to another object at that address).

Unfortunately, we pushed too hard and punched a hole all the way through the required syntax. I need to write a paper to fix this in the Standard so we can use both `==` and `!=` (as intended) even to compare things-that-convert-to-pointers**.

**This is also a lie, at least in portable terms: we can use the operators even for pointers only because they provide a total ordering for pointers on our platform. 

TLDR: the intended design is that requiring `invocable<ranges::equal_to, T, U>` gives me the capability to compare `T`s with `U`s (and `T`'s with `T`'s, and `U`s with `U`s) with `==` and `!=` on a platform for which those operators totally order pointers.

Reply:
(pause for breath)

Oh, and you should clarify what you mean by "Same question", since I'm now suspicious that there's another operation for which you are questioning constraint coverage.

Reply:
Apparently I made a mistake *again* and submitted a review without finalizing a comment edit, or something (I am pretty sure I am doing something wrong but haven't noticed it happening). Yeah, that's what I was asking. The two occurrences were `remove` and `remove_copy` testing `!= _Val`, nothing mysterious.